langlib/lang.table/src/main/ballerina/table.bal

```diff
@@ -110,6 +182,13 @@ public isolated function add(table<MapType> t, MapType val) = @java:Method {
 
 # Applies a function to each member of a table and returns a table of the result.
 #
+# ```ballerina
+# table [
+#     {id: 1, maths: 78, physics: 70},
+#     {id: 2, maths: 83, physics: 80}
+# ].map(student => {id: student.id, avg: (student.maths + student.physics) / 2}) ⇒ [{"id":1,"avg":74},{"id":2,"avg":81}]
```

Comment:
With arrow functions we decided to use a separate variable for the value since the type for the parameter is inferred. Please check and update wherever applicable.

Reply:
Can you give an example?

Reply:
@MaryamZi

Reply:
Can you elaborate on what's not clear?

Reply:
```
table<record {| int id; int maths; int physics; |}> students = table [
    {id: 1, maths: 78, physics: 70},
    {id: 2, maths: 83, physics: 80}
];
students.map(student => {id: student.id, avg: (student.maths + student.physics) / 2}) ⇒ [{"id":1,"avg":74},{"id":2,"avg":81}]
```

Reply:
Let's use "math" instead of "maths" btw.